app/src/main/java/org/mozilla/fenix/searchdialog/SearchDialogFragment.kt

```diff
@@ -111,38 +129,111 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        if (view.context.settings().toolbarPosition == ToolbarPosition.BOTTOM) {
-            ConstraintSet().apply {
-                clone(search_wrapper)
-
-                clear(toolbar.id, TOP)
-                connect(toolbar.id, BOTTOM, PARENT_ID, BOTTOM)
-
-                clear(awesomeBar.id, TOP)
-                clear(awesomeBar.id, BOTTOM)
-                connect(awesomeBar.id, TOP, PARENT_ID, TOP)
-                connect(awesomeBar.id, BOTTOM, toolbar.id, TOP)
-
-                applyTo(search_wrapper)
-            }
-        }
+        setupConstraints(view)
 
         search_wrapper.setOnClickListener {
             it.hideKeyboard()
             dismissAllowingStateLoss()
         }
 
+        view.search_engines_shortcut_button.setOnClickListener {
+            interactor.onSearchShortcutsButtonClicked()
+        }
+
+        search_scan_button.setOnClickListener {
```

Comment:
In all other case we now first check if the device has a camera to avoid crashing (or displaying errors) on TV boxes etc. (The number of devices without camera is surprisingly high on release :)).

We have a utility function `context.hasCamera` we can use.

Reply:
Ah good catch!